conan/tools/intel/intel_cc.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+    Intel generator module for oneAPI Toolkits.
+
+    For simplicity and clarity, Intel informally refers to some of the terms in this document,
+    as listed below:
+
+        - ICX - Intel oneAPI DPC++/C++ Compiler
+        - ICC Classic - Intel C++ Compiler Classic
+        - DPCPP - Intel oneAPI DPC++/C++ Compiler
+
+    DPCPP is built upon ICX as the underlying C++ Compiler, therefore most of this information
+    also applies to DPCPP.
+
+    Intel oneAPI Toolkit (DPC++/C++ Compiler)
+        - Versioning: https://software.intel.com/content/www/us/en/develop/articles/oneapi-toolkit-version-to-compiler-version-mapping.html
+        - Compiler: https://software.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top.html
+"""
+import os
+import platform
+import textwrap
+
+from conan.tools.env.environment import create_env_script
+from conans.client.tools.win import is_win64, _system_registry_key
+from conans.errors import ConanException
+
+
+def _is_using_intel_oneapi(compiler_version):
+    """Check if the Intel compiler to be used belongs to Intel oneAPI
+
+    Note: Intel oneAPI Toolkit first version is 2021.1
+    """
+    return int(compiler_version.split(".")[0]) >= 2021
+
+
+def _get_intel_cc_installation_path():
```

Comment:
I would remove this, to be honest. I'm sure it works and it is a great job but looks like something hard to maintain and probably can differ in different Linux distributions/package managers. I think the users using that new Intel compiler will be glad to introduce the config. We can mention in the documentation the typical routes.
WDYT @franramirez688 ?

Reply:
Yeah, you're absolutely right but I don't know what to answer... For us, it will be the simplest solution for sure but I think it does not matter if we "suggest" two different default paths if the user did not, does it? In terms of maintenance, I think we can skip it saying that you can put any path through the `[conf]` so we won't have to be aware of the Intel default path changes, right?